src/Common/Commands.Common.Authentication/HttpClientOperationsFactory.cs

```diff
@@ -0,0 +1,116 @@
+            {
+                var response = await SafeSendRequestAsync(new HttpRequestMessage(HttpMethod.Get, requestUri), token);
+                var stringContent = await response.Content.ReadAsStringAsync();
+                return JsonConvert.DeserializeObject<T>(stringContent);
+            }
+
+            public async Task<bool> HeadAsync(string requestUri, CancellationToken token)
+            {
+                var response = await SafeSendRequestAsync(new HttpRequestMessage(HttpMethod.Head, requestUri), token);
+                return response.IsSuccessStatusCode;
+            }
+
+            public async Task<IEnumerable<T>> ListAsync(string requestUri, CancellationToken token)
+            {
+                var response = await SafeSendRequestAsync(new HttpRequestMessage(HttpMethod.Get, requestUri), token);
+                var stringContent = await response.Content.ReadAsStringAsync();
+                return JsonConvert.DeserializeObject<List<T>>(stringContent);
+            }
+
+            public async Task<T> PutAsync(string requestUri, T payload, CancellationToken token)
+            {
+                var requestContent = JsonConvert.SerializeObject(payload);
+                var request = new HttpRequestMessage(HttpMethod.Put, requestUri);
+                request.Content = new StringContent(requestContent);
+                var response = await SafeSendRequestAsync(request, token);
+                var stringContent = await response.Content.ReadAsStringAsync();
+                return JsonConvert.DeserializeObject<T>(stringContent);
+            }
+
+            public IHttpOperations<T> WithHeader(string name, IEnumerable<string> value)
+            {
+                ServiceClientTracing.Information("[HttpClientOperations]: Adding Header '{0}'", name);
+                _headers.Add(name, value);
+                return this;
+            }
+
+            async Task<HttpResponseMessage> SafeSendRequestAsync(HttpRequestMessage request, CancellationToken token)
+            {
+                var invocationId = string.Format("[HttpClientOperations {0}]", ServiceClientTracing.NextInvocationId);
+                foreach (var header in _headers)
+                {
+                    request.Headers.Add(header.Key, header.Value);
+                }
+
+                ServiceClientTracing.SendRequest(invocationId, request);
+                var response = await _client.SendAsync(request, token);
+                ServiceClientTracing.ReceiveResponse(invocationId, response);
+                if (!response.IsSuccessStatusCode)
+                {
+                    var exception = new HttpRequestException(string.Format("Unexpected response status code '{0}' received for request '{{{1} {2}}} Body: {{{3}}}", 
+                        response.StatusCode, request.Method, request.RequestUri, response.Content.ReadAsStringAsync().GetAwaiter().GetResult()));
+                    ServiceClientTracing.Error(invocationId, exception);
```

Comment:
Any reason to not use string interpolation?

Reply:
No, but this should probably be a string resource, anyway